src/adapter/src/catalog/storage/stash.rs

```diff
@@ -561,20 +565,43 @@ pub async fn initialize(
     CONFIG_COLLECTION
         .initialize(
             tx,
-            [(
-                proto::ConfigKey {
-                    key: USER_VERSION.to_string(),
-                },
-                proto::ConfigValue {
-                    value: STASH_VERSION,
-                },
-            )],
+            [
+                (
+                    proto::ConfigKey {
+                        key: USER_VERSION.to_string(),
+                    },
+                    proto::ConfigValue {
+                        value: STASH_VERSION,
+                    },
+                ),
+                (
+                    proto::ConfigKey {
+                        key: DEPLOY_GENERATION.to_string(),
```

Comment:
Putting this into the initial migration scares me a bit because the generation changes so often. I'd rather that it's always explicitly checked and set after the migrations are done.

Reply:
i'm not sure i understand the worry. Is it that we might have a bug where we don't set the deploy generation later and it'll always appear as zero?

It's checked before we do `Connection::open` which can set the deploy generation, we only go through the initialize code path when it hasn't been initialized. The normal path is to call `stash.upgrade()` and then `conn.persist_deploy_generation` in `Connection::open`.

Reply:
Ok, SGTM.